java/src/main/java/org/rocksdb/FlinkCompactionFilter.java

```diff
@@ -0,0 +1,142 @@
+//  Copyright (c) 2017-present, Facebook, Inc.  All rights reserved.
+//  This source code is licensed under both the GPLv2 (found in the
+//  COPYING file in the root directory) and Apache 2.0 License
+//  (found in the LICENSE.Apache file in the root directory).
+
+package org.rocksdb;
+
+/**
+ * Just a Java wrapper around FlinkCompactionFilter implemented in C++.
+ *
+ * Note: this compaction filter is a special implementation, designed for usage only in Apache Flink project.
+ */
+public class FlinkCompactionFilter
+    extends AbstractCompactionFilter<Slice> {
+  public enum StateType {
+    // WARNING!!! Do not change the order of enum entries as it is important for jni translation
+    Disabled,
+    Value,
+    List
+  }
+
+  public FlinkCompactionFilter(ConfigHolder configHolder) {
+    this(configHolder, null);
+  }
+
+  public FlinkCompactionFilter(ConfigHolder configHolder, Logger logger) {
+    super(createNewFlinkCompactionFilter0(configHolder.nativeHandle_, logger == null ? 0 : logger.nativeHandle_));
+  }
+
+  private native static long createNewFlinkCompactionFilter0(long configHolderHandle, long loggerHandle);
+  private native static long createNewFlinkCompactionFilterConfigHolder();
+  private native static void disposeFlinkCompactionFilterConfigHolder(long configHolderHandle);
+  private native static long configureFlinkCompactionFilter(
+          long configHolderHandle, int stateType, int timestampOffset, long ttl, boolean useSystemTime,
+          int fixedElementLength, ListElementIterFactory listElementIterFactory);
+  private native static long setCurrentTimestamp(
+          long configHolderHandle, long currentTimestamp);
+
+  public interface ListElementIter {
+    void setListBytes(byte[] list);
+    int nextOffset(int currentOffset);
```

Comment:
Looks like this iterator abstraction comes at the cost of calling from C++ into Java per element. I think a bulk operation that returns the offset of all expired elements might be more straight forward and more performant.